TheBTeam.ConsoleApp/ConsoleFactory.cs

```diff
@@ -337,6 +357,122 @@ public static Transaction ApplyTransaction(Transaction transaction, User user)
+            bool exit = false;
+            while (exit == false)
+            {
+                Console.WriteLine(options, "\n");
+                string choice = Console.ReadLine();
+                if (choice == "1")
+                {
+                    tempuser.LastName = GetStringInput("Last Name", MinNameLength);
+                }
+                else if (choice == "2")
+                {
+                    tempuser.Age = GetIntInput("Age", MinAge, MaxAge);
+                }
+                else if (choice == "3")
+                {
+                    tempuser.Company = GetCompany();
+                }
+                else if (choice == "4")
+                {
+                    tempuser.Gender = GetGender();
+                }
+                else if (choice == "5")
+                {
+                    tempuser.Phone = GetPhoneNumber();
+                }
+                else if (choice == "6")
+                {
+                    tempuser.Balance = GetDecimalInput("current balance");
+                }
+                else if (choice == "7")
+                {
+                    tempuser.Currency = GetCurrency();
+                }
+                else if (choice == "8")
+                {
+                    Console.WriteLine("S to save, X to cancel\n");
+                    bool choiceSubmitted = false;
+                    while (choiceSubmitted == false)
+                    {
+                        string key = Console.ReadKey().Key.ToString();
+                        if (key.ToUpper() == "S")
+                        {
+                            user.LastName = tempuser.LastName;
+                            user.Age = tempuser.Age;
+                            user.Company = tempuser.Company;
+                            user.Gender = tempuser.Gender;
+                            user.Phone = tempuser.Phone;
+                            user.Currency = tempuser.Currency;
+                            user.Balance = tempuser.Balance;
+                            Console.WriteLine("\nSaved");
+                            choiceSubmitted = true;
+                            exit = true;
+                        }
+
+                        else if (key.ToUpper() == "X")
+                        {
+                            Console.WriteLine("\nCancelled");
+                            choiceSubmitted = true;
+                            exit = true;
+
+                        }
+                        else
+                        {
+                            Console.WriteLine("Select X to abort or S to save please\n");
+                        }
+                    }
+                }
+            }
+            return tempuser;
+        }
+        public static Transaction EditTransaction(Transaction transaction)
+        {
+            var tempTransaction = new Transaction(transaction.User, transaction.Type, transaction.Category, transaction.Currency, transaction.Amount);
```

Comment:
Moim zdaniem efektywniej było by robić zmiany bezpośrednio na transaction, ale jest to ciekawy pomysł z tymczasową zmienną transaction